packages/flutter_tools/bin/xcode_backend.dart

```diff
@@ -349,32 +351,70 @@ class Context {
       return;
     }
 
-    if (verbose) {
-      print('♦ Copying native assets from $nativeAssetsPath.');
-    }
-    for (final FileSystemEntity entity in nativeAssetsDir.listSync()) {
-      if (entity is Directory) {
-        final String? frameworkName = parseFrameworkNameFromDirectory(entity);
-        if (frameworkName != null) {
-          runRsync(
-            extraArgs: <String>[
-              '--filter',
-              '- native_assets.yaml',
-              '--filter',
-              '- native_assets.json',
-            ],
-            entity.path,
-            xcodeFrameworksDir,
-          );
-          if (codesign && expandedCodeSignIdentity != null) {
-            _codesignFramework(
-              expandedCodeSignIdentity,
-              '$xcodeFrameworksDir/$frameworkName.framework/$frameworkName',
+    final Set<String> referencedFrameworks = {};
+    final nativeAssetsJson = File(
+      '$xcodeFrameworksDir/App.framework/flutter_assets/NativeAssetsManifest.json',
+    );
+    if (!nativeAssetsJson.existsSync()) {
+      if (verbose) {
+        print("♦ No native assets to bundle. ${nativeAssetsJson.path} doesn't exist.");
+      }
+      return;
+    }
+    // NativeAssetsManifest.json looks like this: {
```

Comment:
Maybe also add a comment to engine/src/flutter/assets/native_assets.cc where the same format is parsed and a comment from there back to here.